src/lib/djerba/extract/r_script_wrapper.py

```diff
@@ -255,38 +256,70 @@ def preprocess_gep(self, gep_path):
                 writer.writerow(row)
         return out_path
 
+    def tab_file_is_empty(self, mavis_path):
+        """
+        Checks if .tab file provided is empty.
+        Returns False is not empty, returns True if empty.
+        """
+        with open(mavis_path, "rt") as file:
+            num_lines = file.readlines()
+            if len(num_lines) > 1:
+                self.is_tab_empty = False
+
     def preprocess_fus(self, mavis_path):
         """
         Extract the FUS file from the .zip archive output by Mavis
         Apply preprocessing and write results to tmp_dir
         Prepend a column with the tumour id
         """
-        zf = zipfile.ZipFile(mavis_path)
-        matched = []
-        for name in zf.namelist():
-            if re.search('mavis_summary_all_.*\.tab$', name):
-                matched.append(name)
-        if len(matched) == 0:
-            msg = "Could not find Mavis summary .tab in "+mavis_path
-            raise RuntimeError(msg)
-        elif len(matched) > 1:
-            msg = "Found more than one Mavis summary .tab file in "+mavis_path
+        # If the tab file is hidden inside a zip file:
+        if re.search("\.zip$", mavis_path):
+            zf = zipfile.ZipFile(mavis_path)
+            matched = []
+            for name in zf.namelist():
+                if re.search('mavis_summary_all_.*\.tab$', name):
+                    matched.append(name)
+            if len(matched) == 0:
+                msg = "Could not find Mavis summary .tab in "+mavis_path
+                raise RuntimeError(msg)
```

Comment:
Also write these errors to the log please, with `self.logger.error(msg)`

Reply:
In addition to raise RuntimeError(msg) (i.e. right before it) or in replacement of RuntimeError(msg)?